lib/storage/data/DataWrapper.js

```diff
@@ -918,7 +920,24 @@ class DataWrapper {
             hashedStream = new MD5Sum();
             value.pipe(hashedStream);
         }
+        if (keyContext.metaHeaders &&
+            keyContext.metaHeaders['x-amz-meta-mdonly'] === 'true') {
+            const b64 = keyContext.metaHeaders['x-amz-meta-md5chksum'];
+            let md5 = null;
+            if (b64) {
+                md5 = new Buffer(b64, 'base64').toString('hex');
+            }
+            const dataRetrievalInfo = {
+                key: keyContext.objectKey,
+                dataStoreName: this.client.clients[location],
+                dataStoreType: this.client.clients[location].clientType,
```

Comment:
will need to perform a check to make sure `this.client.client` exists.
`file / mem` client types do not have the `client.client` map

Reply:
Will do, thank you!